nodewatcher/nodewatcher.py

```diff
@@ -235,44 +243,38 @@ def _poll_instance_status(config, scheduler_module, asg_name, hostname, instance
     :param hostname: current hostname
     :param instance_id: current instance id
     """
+    log.info("Waiting for stack %s to be ready", config.stack_name)
+    _wait_for_stack_ready(config.stack_name, config.region, config.proxy_config)
```

Comment:
You moved the `_wait_for_stack_ready` outside the main loop but it can be a problem during an update because the current behaviour is that the compute nodes (and the nodewatcher daemons) remain running during an update, so the loop should be "paused" waiting for the UPDATE_*COMPLETE status.

We can move this function outside when we'll force the compute nodes to scale-down during an update.

Reply:
very good point! I'll move this back into the main loop!

Reply:
We agreed that during an update a node not performing any task is still considered idle hence the termination of such nodes is aligned with the expected behaviour. Leaving it as it is.